docs/data/migration/migration-pickers-v5/migration-pickers-v5.md

```diff
@@ -304,3 +304,93 @@ Component name changes are also reflected in `themeAugmentation`:
    },
  });
 ```
+
+### Replace `toolbar` props by slot
+
+- The `ToolbarComponent` has been replaced by a `Toolbar` component slot. You can find more information about this pattern in the [MUI Base documentation](https://mui.com/base/getting-started/usage/#shared-props):
+
+  ```diff
+   // Same on all other pickers
+   <DatePicker
+  -  ToolbarComponent: MyToolbar,
+  +  components={{ Toolbar: MyToolbar }}
+   />
+  ```
+
+- The `toolbarPlaceholder` and `toolbarFormat` props have been moved to the `toolbar` component props slot:
+
+  ```diff
+   // Same on all other pickers
+   <DatePicker
+  -  toolbarPlaceholder="__"
+  -  toolbarFormat="DD / MM / YYYY"
+  +  componentsProps={{
+  +    toolbar: {
+  +      toolbarPlaceholder: "__",
+  +      toolbarFormat: "DD / MM / YYYY",
+  +    }
+  +  }}
+   />
+  ```
+
+- The `toolbarTitle` prop has been moved to the localization object:
+
+  ```diff
+   // Same on all other pickers
+   <DatePicker
+  -  toolbarTitle="Title"
+  +  localeText={{ toolbarTitle: "Title" }}
+   />
+  ```
+
+- The toolbar related translation keys have been renamed to better fit their usage:
+
+  ```diff
+   // Same on all other pickers
+   <DatePicker
+    localeText={{
+  -    datePickerDefaultToolbarTitle: 'Date Picker',
+  +    datePickerToolbarTitle: 'Date Picker',
+
+  -    timePickerDefaultToolbarTitle: 'Time Picker',
+  +    timePickerToolbarTitle: 'Time Picker',
+
+  -    dateTimePickerDefaultToolbarTitle: 'Date Time Picker',
+  +    dateTimePickerToolbarTitle: 'Date Time Picker',
+
+  -    dateRangePickerDefaultToolbarTitle: 'Date Range Picker',
+  +    dateRangePickerToolbarTitle: 'Date Range Picker',
+    }}
+   />
+  ```
+
+- The `onChange` / `openView` props on the toolbar have been renamed to `onViewChange` / `view`
+
+### Replace `tabs` props
+
+- The `hideTabs`, `dateRangeIcon`, and `timeIcon` props have been moved to `tabs` component props:
+
+  ```diff
+   // Same on all other Date Time picker variations
+   <DateTimePicker
+  -  hideTabs={false}
+  -  dateRangeIcon={<LightModeIcon />}
+  -  timeIcon={<AcUnitIcon />}
+  +  componentsProps={{
+  +    tabs: {
+  +      hidden: false,
+  +      dateRangeIcon: <LightModeIcon />,
```

Comment:
This icon name is a bit strange to me.
What do you think about renaming it to `dateIcon` to be more in line with `timeIcon` @alexfauquette @flaviendelangle ? 🤔

Reply:
The name of the prop come from the name of the default icon
But I do agree that in the context of the picker it's very confusing

Reply:
+1 for changing to `dateIcon` which would feel more natural. Better having a meaning full prop naming than to match the default icon name

Reply:
Thank you for the comments.
I've [created a PR](https://github.com/mui/mui-x/pull/7024) to handle it.
Just waiting for this one to be merged firstly. 👌